src/Analyzer/Passes/AnyFunctionPass.cpp

```diff
@@ -0,0 +1,139 @@
+            ///     └────────┴─────────────────────────────────────────┘
+            if (argument->as<LambdaNode>())
+                return false;
+
+            if (argument->as<ArrayJoinNode>())
+                return false;
+
+            if (const auto * inside_function = argument->as<FunctionNode>())
+            {
+                if (!canRewrite(inside_function))
+                    return false;
+            }
+        }
+
+        return true;
+    }
+
+public:
+    using Base = InDepthQueryTreeVisitor<AnyFunctionVisitor>;
+    using Base::Base;
+
+    void visitImpl(QueryTreeNodePtr & node)
+    {
+        auto * function_node = node->as<FunctionNode>();
+        if (!function_node)
+            return;
+
+        /// check function is any
+        const auto & function_name = function_node->getFunctionName();
+        if (!(function_name == "any" || function_name == "anyLast"))
+            return;
+
+        auto & arguments = function_node->getArguments().getNodes();
+        if (arguments.size() != 1)
+            return;
+
+        auto * inside_function_node = arguments[0]->as<FunctionNode>();
+
+        /// check argument is a function and can not be arrayJoin or lambda
+        if (!inside_function_node)
+            return;
+
+        /// check arguments can not contain arrayJoin or lambda
+        if (!canRewrite(inside_function_node))
+            return;
+
+        auto & inside_arguments = inside_function_node->getArguments().getNodes();
+
+        /// case any(f())
+        if (inside_arguments.empty())
+            return;
+
+        if (rewritten.contains(node.get()))
+        {
+            node = rewritten.at(node.get());
+            return;
+        }
+
+        /// checking done, rewrite function
+        bool pushed = false;
+        for (auto & inside_argument : inside_arguments)
+        {
+            if (inside_argument->as<ConstantNode>()) /// skip constant node
+                break;
+
+            AggregateFunctionProperties properties;
+            auto aggregate_function = AggregateFunctionFactory::instance().get(function_name, {inside_argument->getResultType()}, {}, properties);
+
+            auto any_function = std::make_shared<FunctionNode>(function_name);
+            any_function->resolveAsAggregateFunction(std::move(aggregate_function));
+
+            auto & any_function_arguments = any_function->getArguments().getNodes();
+            any_function_arguments.push_back(std::move(inside_argument));
+
+            inside_argument = std::move(any_function);
+            pushed = true;
+        }
+
+        if (pushed)
+        {
+            rewritten.insert({node.get(), arguments[0]});
+            node = arguments[0];
+        }
+    }
+
+private:
+    /// After query analysis alias will be rewritten to QueryTreeNode
+    /// whose memory address is same with the original one.
+    /// So we can reuse the rewritten one.
+    std::unordered_map<IQueryTreeNode *, QueryTreeNodePtr > rewritten;
```

Comment:
I don't think it would optimize anything. I'd get rid of this field.

Reply:
After query analysis alias will be rewritten to QueryTreeNode whose memory address is same with the original one. So we can reuse the rewritten one.

For example: select anyLast(number * 3) AS x, x ...， the two nodes `x` share the same object, when we try to rewrite the first one, the second one will be modified, and when we try wo rewrite the second one,  we wil get a wrong result.

Reply:
@novikd when you have time, please help review the PR.